src/lib/comms/sol-network-impl-linux.c

```diff
@@ -353,7 +353,7 @@ sol_network_start_netlink(void)
     network->nl_socket = socket(AF_NETLINK, SOCK_RAW | SOCK_CLOEXEC | SOCK_NONBLOCK, NETLINK_ROUTE);
     if (network->nl_socket < 0) {
         SOL_WRN("failed to create netlink socket, cannot listen to events or manage the links!");
-        return -errno;
+        return -EBADMSG;
```

Comment:
In this case returning the `-errno` is valid. The `socket()` will set the errno with the correctly value if something goes wrong